src/snowflake/snowpark/column.py

```diff
@@ -225,108 +238,339 @@ class Column:
     This class has methods for the most frequently used column transformations and operators. Module :mod:`snowflake.snowpark.functions` defines many functions to transform columns.
     """
 
+    # NOTE: For now assume Expression instances can be safely ignored when building AST
+    #       Expression logic can be eliminated entirely once phase 0 is integrated
+    #       Currently a breaking example can be created using the Column.isin method as it does not build the AST.
+    #       For example, running: df.filter(col("A").isin(1, 2, 3) & col("B")) would fail since the boolean operator
+    #       '&' would try to construct an AST using that of the new col("A").isin(1, 2, 3) column (which we currently
+    #       don't fill if the only argument provided in the Column constructor is 'expr1' of type Expression)
     def __init__(
-        self, expr1: Union[str, Expression], expr2: Optional[str] = None
+        self,
+        expr1: Union[str, Expression],
+        expr2: Optional[str] = None,
+        _ast: Optional[proto.Expr] = None,
+        _emit_ast: bool = True,
     ) -> None:
+        self._ast = _ast
+
         if expr2 is not None:
-            if isinstance(expr1, str) and isinstance(expr2, str):
-                if expr2 == "*":
-                    self._expression = Star([], df_alias=expr1)
-                else:
-                    self._expression = UnresolvedAttribute(
-                        quote_name(expr2), df_alias=expr1
-                    )
-            else:
+            if not (isinstance(expr1, str) and isinstance(expr2, str)):
```

Comment:
we should not change existing code in this PR. We should limit the scope to add new ast logic only.

Reply:
changes were reviewed by snowpark python team.